src/ComboBox/index.jsx

```diff
@@ -108,583 +104,547 @@ function optionsFormatted( filteredOptionsIds, originalOptions )
-            activeOption    : undefined,
-            filteredOptions : undefined,
-            flatOptions     : undefined,
-            id              : undefined,
-            isOpen          : undefined,
-            options         : undefined,
-            searchValue     : undefined,
-            selection       : isMultiselect ?
-                castArray( defaultValue ) : defaultValue,
-        };
-
-        this.inputRef     = React.createRef();
-        this.scrollBoxRef = React.createRef();
-
-        this.handleBlur            = this.handleBlur.bind( this );
-        this.handleChangeInput     = this.handleChangeInput.bind( this );
-        this.handleClick           = this.handleClick.bind( this );
-        this.handleClickClose      = this.handleClickClose.bind( this );
-        this.handleClickIcon       = this.handleClickIcon.bind( this );
-        this.handleClickOption     = this.handleClickOption.bind( this );
-        this.handleFocus           = this.handleFocus.bind( this );
-        this.handleKeyDown         = this.handleKeyDown.bind( this );
-        this.handleMouseOutOption  = this.handleMouseOutOption.bind( this );
-        this.handleMouseOverOption = this.handleMouseOverOption.bind( this );
-    }
-
-    static getDerivedStateFromProps( props, state )
-    {
-        let { flatOptions } = state;
-
-        const options = normalizeOptions( props.options ) ||
-            state.options || [];
-
-        if ( options && options !== state.options )
+        if ( !value )
         {
-            flatOptions = options.flatMap( o => o.options || o );
+            setSelection( newValue );
         }
+    };
 
-        let { selection } = state;
-        if ( props.value )
-        {
-            selection = props.value;
-        }
+    return [ validatedValue || validatedSelection, setter ];
+};
 
-        if ( props.isMultiselect )
-        {
-            selection = castArray( selection );
-        }
+const componentName = 'ComboBox';
 
-        const filteredOptions = state.searchValue && (
-            flatOptions.filter( ( { text } ) => (
-                text.match( new RegExp(
-                    escapeRegExp( state.searchValue ),
-                    'i',
-                ) )
+const ComboBox = props =>
+{
+    const cssMap = useTheme( componentName, props );
+    const [ stateActiveOption, setActiveOption ] = useState( undefined );
+    const [ isOpen, setIsOpen ] = useState( undefined );
+    const [ searchValue, setSearchValue ] = useState( undefined );
+
+    const inputRef = useRef( null );
+    const scrollBoxRef = useRef( null );
+
+    const {
+        container,
+        defaultValue,
+        dropdownPlaceholder,
+        hasError,
+        inputPlaceholder,
+        isDisabled,
+        isMultiselect,
+        isReadOnly,
+        isSearchable,
+        onChange,
+        onChangeInput,
+        options,
+        value,
+    } = props;
+
+    const [ optionSelected, setOptionSelected ] = useSelection(
```

Comment:
Can we call it `selection` and `setSelection` instead? `optionSelected` implies a single option but it could be multiple